README.md

```diff
@@ -3,45 +3,245 @@
+    in the same JSON-LD document using a wide variety of
+    [*cryptographic suites*][cryptosuite].
 
-We are setting up a process to accept contributions. Please feel free to open
-issues or PRs in the interim, but we cannot merge external changes until this
-process is in place.
+[dids]: <https://www.w3.org/TR/did-core/>
+[didkit]: <https://github.com/spruceid/didkit>
+[vc-data-model]: <https://www.w3.org/TR/vc-data-model/>
+[jwt]: <https://www.rfc-editor.org/rfc/rfc7519>
+[jws]: <https://www.rfc-editor.org/rfc/rfc7515>
+[jwt-vc]: <https://www.w3.org/TR/vc-data-model/#json-web-token>
+[data-integrity]: <https://www.w3.org/TR/vc-data-integrity/>
+[cryptosuite]: <https://www.w3.org/TR/vc-data-integrity/#dfn-cryptosuite>
+
+## Basic Usage
 
-## Install
+SSI provides various functions to parse, verify, create and sign various
+kind of claims. This section shows how to use these functions in combination
+with JSON Web Signatures (or Tokens) and Verifiable Credentials.
 
-### Crates.io
+### Verification
 
+The simplest type of claim to load and verify is probably JSON Web
+Signatures (JWSs), often use to encode JSON Web Tokens (JWTs). To represent
+such claims SSI provides the `CompactJWSString` type representing a JWS
+in compact textual form. One can load a JWS using `from_string` and verify
+it using `verify`.
+
+
+```rust
+use ssi::prelude::*;
+
+// Load a JWT from the file system.
+let jwt = CompactJWSString::from_string(
+  std::fs::read_to_string("examples/files/claims.jwt")
+  .expect("unable to load JWT")
+).expect("invalid JWS");
+
+// Setup a verification method resolver, in charge of retrieving the
+// public key used to sign the JWT.
+// Here we use the example `ExampleDIDResolver` resolver, enabled with the
+// `example` feature.
+let vm_resolver = ExampleDIDResolver::default().with_default_options();
+
+// Verify the JWT.
+assert!(jwt.verify::<AnyJwkMethod>(&vm_resolver).await.expect("verification failed"))
 ```
-ssi = "0.4"
+
+#### Verifiable Credentials
+
+Verifiable Credential are much more complex as they require interpreting
+the input claims and proofs, such as Data-Integrity proofs as Linked-Data
+using JSON-LD. This operation is highly configurable. SSI provide
+functions exposing various levels of implementation details that you can
+tweak as needed. The simplest of them is `any_credential_from_json_str`
+that will simply load a VC from a string, assuming it is signed using
+any Data-Integrity proof supported by SSI.
+
+
+```rust
+use ssi::prelude::*;
+
+let vc = any_credential_from_json_str(
+  &std::fs::read_to_string("examples/files/vc.jsonld")
+  .expect("unable to load VC")
+).await.expect("invalid VC");
+
+// Setup a verification method resolver, in charge of retrieving the
+// public key used to sign the JWT.
+let vm_resolver = ExampleDIDResolver::default().with_default_options();
+
+assert!(vc.verify(&vm_resolver).await.expect("verification failed").is_valid());
```

Comment:
snippet runs successfully as example script, with `examples` feature and wrapped in a main function